docs/dummyvm/src/slots.rs

```diff
@@ -0,0 +1,233 @@
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum DummyVMSlot {
+    Simple(SimpleSlot),
+    #[cfg(target_pointer_width = "64")]
+    Compressed(only_64_bit::CompressedOopSlot),
+    Offset(OffsetSlot),
+    Tagged(TaggedSlot),
+}
+
+unsafe impl Send for DummyVMSlot {}
+
+impl Slot for DummyVMSlot {
+    fn load(&self) -> Option<ObjectReference> {
+        match self {
+            DummyVMSlot::Simple(e) => e.load(),
+            #[cfg(target_pointer_width = "64")]
+            DummyVMSlot::Compressed(e) => e.load(),
+            DummyVMSlot::Offset(e) => e.load(),
+            DummyVMSlot::Tagged(e) => e.load(),
+        }
+    }
+
+    fn store(&self, object: ObjectReference) {
+        match self {
+            DummyVMSlot::Simple(e) => e.store(object),
+            #[cfg(target_pointer_width = "64")]
+            DummyVMSlot::Compressed(e) => e.store(object),
+            DummyVMSlot::Offset(e) => e.store(object),
+            DummyVMSlot::Tagged(e) => e.store(object),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct DummyVMMemorySlice(*mut [ObjectReference]);
+
+unsafe impl Send for DummyVMMemorySlice {}
+
+impl MemorySlice for DummyVMMemorySlice {
+    type SlotType = DummyVMSlot;
+    type SlotIterator = DummyVMMemorySliceIterator;
+
+    fn iter_slots(&self) -> Self::SlotIterator {
+        DummyVMMemorySliceIterator {
+            cursor: unsafe { (*self.0).as_mut_ptr_range().start },
+            limit: unsafe { (*self.0).as_mut_ptr_range().end },
+        }
+    }
+
+    fn object(&self) -> Option<ObjectReference> {
+        None
+    }
+
+    fn start(&self) -> Address {
+        Address::from_ptr(unsafe { (*self.0).as_ptr_range().start })
+    }
+
+    fn bytes(&self) -> usize {
+        unsafe { std::mem::size_of_val(&*self.0) }
+    }
+
+    fn copy(src: &Self, tgt: &Self) {
+        debug_assert_eq!(src.bytes(), tgt.bytes());
+        debug_assert_eq!(
+            src.bytes() & ((1 << LOG_BYTES_IN_ADDRESS) - 1),
+            0,
+            "bytes are not a multiple of words"
+        );
+        // Raw memory copy
+        unsafe {
+            let words = tgt.bytes() >> LOG_BYTES_IN_ADDRESS;
+            let src = src.start().to_ptr::<usize>();
+            let tgt = tgt.start().to_mut_ptr::<usize>();
+            std::ptr::copy(src, tgt, words)
+        }
+    }
+}
+
+pub struct DummyVMMemorySliceIterator {
+    cursor: *mut ObjectReference,
+    limit: *mut ObjectReference,
+}
+
+impl Iterator for DummyVMMemorySliceIterator {
+    type Item = DummyVMSlot;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.cursor >= self.limit {
+            None
+        } else {
+            let edge = self.cursor;
+            self.cursor = unsafe { self.cursor.add(1) };
+            Some(DummyVMSlot::Simple(SimpleSlot::from_address(
+                Address::from_ptr(edge),
+            )))
+        }
+    }
+}
+
+/// Compressed OOP edge only makes sense on 64-bit architectures.
```

Comment:
There are other occurrences of "edge" in this file.  It's helpful to do a find/replace.

Reply:
Theses slots code are removed.